### DISCUSSION TITLE
Shelly H&T

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#13630] [Ideas]
It would be amazing to get Tasmota support for the Shelly H&T since there's an USB-power supply for it now! I searched previous issues but one was closed because there was only battery power at the time, and another was closed due to ignoring the submission template. https://www.shelly.si/en/accessories-for-shelly-products/19-shelly-ht-usb-power-supply.html  

According to articles such as this one it should be based on the ESP8266, but I could probably get some pictures of the board if really needed. https://www.cnx-software.com/2018/10/30/shelly-ht-battery-powered-esp8266-wifi-temperature-humidity-sensor/

**Comment by sfromis** [comment]
From what others have looked at, it appears that the unknown (?) temperature sensor is handled via another MCU, with an unknown (?) communication with the ESP8266. To get it supported, someone would have to decode the protocol, and do what would be needed to emulate it. Or reprogram that other MCU, or mess with the hardware.

**Reply by sfromis** [reply]
If anyone is interested, on Discord Flanagun posted this board image:
![Shelly H T board](https://user-images.githubusercontent.com/47082390/147769328-8ede92ae-960e-49c3-a40b-07cb8b404ee1.jpg)

**Comment by SumDum** [comment]
My findings so far:
The sensor is a GXHT30.  These's a chinese language data sheet here
 [http://www.gxcas.com/public/en/public/html/index.html#/proDetail?pid=23&id=87](url) 
No luck at all identifying the intermediate MCU.  So far as I can make out the markings are:
`LO31U6
GQ21115
VGCHNX
OS2`
The ESP flash size is 2M.
I flashed one with tasmota.bin & tasmota-minimal 10.1.0 just to see what would happen. Neither boot up sufficient to connect to WiFi or even respond via serial.  The LED just flashes (fast) continually, perhaps stuck in a boot loop.
My only real success has been establishing that the battery powered ones work fine off a protected 16340 lithium cell so there's no need to worry about disposables nor tie it to a USB wall wart.  The H&T has a buck/boost regulator which delivers stable 3.3v across the lithium operating range. 

That's probably the limit of what I'm going to be able to determine.

**Reply by barbudor** [reply]
Flashing with tasmota-minimal is a very bad idea in general.
Never ever flash a new device with tasmota-minimal as it is dedicated to be only use in upgrade process with certain conditions.
If you are looking for a small workable tasmota binary, use tasmota-lite.

This being said, as standard tasmota.bin doesn't boot completely, this is generally the case in battery powered sensors where a low power MCU is in control and wakes up the ESP only when there is wifi transmission required.
To make it work with Tasmota, some hardware tinkering may be required. But before a detailled analysis of what happens between the MCU and the ESP using the original firmware would be needed.

If you want a H&T sensor that is USB powered, just get a NodeMCU with a BME280 or an HDC1080.
Even with battery powered you can do better with a modified NodeMCU/Mini-D1 and using DeepSleep.

**Reply by SumDum** [reply]
Noted about tasmota-minimal thanks, I won't do that again!  It survived this fool's bumbling & is back running stock firmware which lacks a great deal of configurability.

**Comment by janknieling** [comment]
Has anyone tried to get tasmota working on the Shelly Plus H&T based on the ESP32 regarding the dataset (https://www.shelly.cloud/en/products/shop/shelly-plus-h-t)?

**Comment by zinob** [comment]
There is a STM32l031g6 MCU on the board, so I guess they use that rather than the ESP-low-power mode. (I found this post while trying to figure out if the H&T had a voltage-regulator so I have no further input really).

**Reply by zinob** [reply]
Oh, and the the XT25F16f chip is an SPI flash-memory (or a folding picnic table, depending on which google result i trust).

**Reply by zinob** [reply]
![image](https://github.com/arendst/Tasmota/assets/3193966/9e293923-d29b-46c7-9bbc-b487f3e735a9)
![image](https://github.com/arendst/Tasmota/assets/3193966/3130234f-8f22-454f-9cb2-817e42c8b51a)
![image](https://github.com/arendst/Tasmota/assets/3193966/793bcf29-d9b6-4923-b84b-9be04ef5d430)

**Comment by pepsi133** [comment]
Have anyone tried flashing the newer 3rd generation Shelly H&T Gen 3? It supposedly has 8MB of flash.
I'm hoping to be able to use it as BLE only sensor and/or customise it otherwise with Tasmota.
Any tips before I commit to burying bulk of those sensors?

**Reply by arcnid** [reply]
I have not tried the 3rd gen but the 1st gen is likely impossible to get working correctly with custom firmware because of the additional STM microcontroller. 

From what I can see online the gen 3 doesn't even expose ports for the device to be flashed with. 

Let me know if you get this working tho lol.

**Reply by Jason2866** [reply]
There will be no support from Tasmota. Reason Tasmota is not designed for Battery operation.
For "just" Humidity & Temperatur there are easy / cheap solutions with Tasmota already possible.
The Shelly device would loose its great benefit. Battery Operation.

**Reply by pepsi133** [reply]
That's a pity. I'm actually going to power them from USB port anyway.
Hence the lack of battery operation is not an issue.

I haven't seen any H&T sensors with e-ink display integrated and such nice form factor
